include/circt/Dialect/HW/HWMiscOps.td

```diff
@@ -53,6 +54,52 @@ def ConstantOp
   let hasVerifier = 1;
 }
 
+def WireOp : HWOp<"wire", [
+    SameOperandsAndResultType,
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>]> {
+  let summary = "Assign a name or symbol to an SSA edge";
+  let description = [{
+    An `hw.wire` is used to assign a human-readable name or a symbol for remote
+    references to an SSA edge. It takes a single operand and returns that
+    its value unchanged as a result. The operation guarantees the following:
+
+    - The value of its operand remains observable under the given human-readable
+      name in any final output, or under the given symbol within the IR.
+    - The users of its result will always observe the operand through the
+      operation itself, meaning that optimizations cannot bypass the wire. This
+      ensures that if the wire's value is *forced*, for example through a
+      Verilog force statement, the forced value will affect all users of the
+      wire in the output.
```

Comment:
Since you have folders for the wire, I think that these are only guaranteed if it has a symbol?

Reply:
I see Mike beat me to it 😄